File: model/src/main/java/net/artux/pda/model/Summary.java
package net.artux.pda.model;

import java.time.Instant;
import java.time.ZoneId;
import java.time.format.DateTimeFormatter;
import java.util.ArrayList;
import java.util.List;

import lombok.Data;

@Data
public class Summary {

    private final String title;
    private final List<UserMessage> messages;

    public Summary() {
        this.title = getCurrentId();
        messages = new ArrayList<>();
    }

    public static String getCurrentId() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy")
                .withZone(ZoneId.systemDefault());
        return formatter.format(Instant.now());
    }

    public void addMessage(UserMessage message) {
        messages.add(message);
    }

}

File: model/src/main/java/net/artux/pda/model/Summary.kt
package net.artux.pda.model

import net.artux.pda.model.chat.UserMessage
import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter

class Summary {
    var title: String = currentId
    var messages: MutableList<UserMessage> = mutableListOf()

    fun addMessage(message: UserMessage) {
        messages.add(message)
    }

    companion object {
        val currentId: String
            get() {
                val formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy")
                    .withZone(ZoneId.systemDefault())
                return formatter.format(Instant.now())
            }
    }
}